Sources/Utility/ArgumentParser.swift

```diff
@@ -278,8 +265,8 @@ final class AnyArgument: ArgumentProtocol, CustomStringConvertible {
     }
 
     /// Initializer for array arguments.
-    init<T>(_ arg: OptionArgument<[T]>) {
-        self.kind = T.self as! ArgumentKind.Type
+    init<T: ArgumentProtocol>(_ arg: T) where T.ArgumentKindTy: Sequence {
```

Comment:
Can't find a way to get it to work with `Array`. That's the reason I used `Sequence`. It has to do with `Array` being a nominal type. Have tried:

```
init<T: ArgumentProtocol>(_ arg: T) where T.ArgumentKindTy: [ArgumentKind] // inheritance from non-named type [ArgumentKind]
init<T: ArgumentProtocol, U: ArgumentKind>(_ arg: T) where T.ArgumentKindTy: [U] // inheritance from non-named type U
```

Let me know if you have other ideas.